gremlin-server/src/test/java/org/apache/tinkerpop/gremlin/server/GremlinDriverIntegrateTest.java

```diff
@@ -612,40 +612,42 @@ public void shouldProcessRequestsOutOfOrder() throws Exception {
         }
     }
 
+    /**
+     * This test validates that the session requests are processed in-order on the server. The order of results
+     * returned to the client might be different though since each result is handled by a different executor thread.
+     */
```

Comment:
just to clarify , by "different executor thread" are you referring to threads on the client being responsible for the un-ordering?

Reply:
Yes. Client side threads. More specifically, in [this](https://github.com/apache/tinkerpop/pull/1347/files#diff-1a4243209a3c0c168ac7e9817b3bca1a2791db382c294130ea85c6f0bcec120cR109) PR we made a change to remove the possibility of blocking the executor loop on client by the application code by passing completion of readComplete future to client's executor threads. For two different requests they might get assigned to different threads in executor pool here and return results at different time.